### PR TITLE
Hide recommended playlist on homepage.

### PR DESCRIPTION
**Pull request by HelloIamarandomperson**
Sorry for not being great at this, new at most of this.

Also is there any chance you can give me a list of things I should add to .gitignore?

## Release Notes

- **New Features**
  - Added an option to hide playlist recommendations from the homepage.
  - Introduced a feature to skip the "continue watching" prompt during video playback.

- **Localization Updates**
  - Added new settings for "playlistRemover" and "skipContinueWatching" in multiple language locales.

- **Bug Fixes**
  - Corrected JSON structure in various localization files to ensure proper parsing.

- **Configuration Changes**
  - Updated configuration options to include the ability to enable or disable hiding playlist recommendations.

### CODE REVIEW

**Comment by VampireChicken12** [comment]
> Sorry for not being great at this, new at most of this.

It's okay that's fine. Sorry that there isn't a contributing guide.  

> Also is there any chance you can give me a list of things I should add to .gitignore?

You shouldn't have to add anything to the .gitignore

**Comment by VampireChicken12** [comment]
@coderabbitai review

**Comment by VampireChicken12** [comment]
@coderabbitai summary

**Comment by VampireChicken12** [comment]
@coderabbitai full review

File: src/features/hidePlaylist/index.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
+import type { Nullable } from "@/src/types";
+import { waitForSpecificMessage } from "@/src/utils/utilities";
+
+let recommendationsObserver: Nullable<MutationObserver> = null;
+let observerDisabled = false;
+
+export async function enableHidePlaylist() {
```
Comment (VampireChicken12) [inline src/features/hidePlaylist/index.ts]: The function name should be enableHidePlaylistRecommendationsFromHomePage

File: src/features/hidePlaylist/index.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      options: { enable_hide_playlist_recommendations_from_home_page },
+    },
+  } = await waitForSpecificMessage("options", "request_data", "content");
+
+  if (!enable_hide_playlist_recommendations_from_home_page) return;
+
+  hideRecommendations();
+  observeHomePageRecommendations();
+}
+
```
Comment (VampireChicken12) [inline src/features/hidePlaylist/index.ts:19]: The function name should be disableHidePlaylistRecommendationsFromHomePage

File: src/pages/content/index.ts [hunk 3]
```diff
@@ -478,7 +477,12 @@ const storageChangeHandler = async (changes: StorageChanges, areaName: string) =
 				volumeBoostEnabled: options.enable_volume_boost,
 				volumeBoostMode: newValue
 			});
-		}
+        },
+		enable_hide_playlist_recommendations_from_home_page: (__oldValue, newValue) => {
+            sendExtensionOnlyMessage("hidePlaylistChange", {
```
Comment (VampireChicken12) [inline src/pages/content/index.ts]: Change hidePlaylistChange to hidePlaylistRecommendationsFromHomePageChange

File: src/pages/content/index.ts [hunk 4]
```diff
@@ -478,7 +477,12 @@ const storageChangeHandler = async (changes: StorageChanges, areaName: string) =
...
+                hidePlaylistEnabled: newValue,
```
Comment (VampireChicken12) [inline src/pages/content/index.ts]: Change hidePlaylistEnabled to hidePlaylistRecommendationsFromHomePageEnabled

File: src/pages/embedded/index.ts [hunk 5]
```diff
@@ -967,6 +969,16 @@ window.addEventListener("DOMContentLoaded", function () {
 					default: {
 						return;
 					}
+					case "hidePlaylistChange": {
```
Comment (VampireChicken12) [inline src/pages/embedded/index.ts]: Change hidePlaylistChange to hidePlaylistRecommendationsFromHomePageChange

File: src/pages/embedded/index.ts [hunk 6]
```diff
@@ -967,6 +969,16 @@ window.addEventListener("DOMContentLoaded", function () {
...
+						const {
+							data: { hidePlaylistEnabled }
+						} = message;
+						if (hidePlaylistEnabled) {
```
Comment (VampireChicken12) [inline src/pages/embedded/index.ts]: Change hidePlaylistEnabled to hidePlaylistRecommendationsFromHomePageEnabled

File: src/types/index.ts [hunk 7]
```diff
@@ -418,8 +418,11 @@ export type ExtensionSendOnlyMessageMappings = {
 	volumeBoostChange: DataResponseMessage<"volumeBoostChange", { volumeBoostEnabled: boolean; volumeBoostMode: VolumeBoostMode }>;
 	automaticallyDisableClosedCaptionsChange: DataResponseMessage<
 		"automaticallyDisableClosedCaptionsChange",
-		{ automaticallyDisableClosedCaptionsEnabled: boolean }
+		{ automaticallyDisableClosedCaptionsEnabled: boolean 
+			
+		}
 	>;
+		hidePlaylistChange: DataResponseMessage<"hidePlaylistChange", { hidePlaylistEnabled: boolean }>;
```
Comment (VampireChicken12) [inline src/types/index.ts]: Change hidePlaylistChange to hidePlaylistRecommendationsFromHomePageChange

File: src/types/index.ts [hunk 8]
```diff
@@ -447,6 +450,10 @@ export type MessageMappings = Prettify<{
 			| SendDataMessage<"send_data", "content", "videoHistoryOne", { video_history_entry: VideoHistoryEntry }>;
 		response: DataResponseMessage<"videoHistoryOne", { video_history_entry: VideoHistoryEntry }>;
 	};
+	enable_hide_playlist_recommendations_from_home_page: {
+        request: RequestDataMessage<"enable_hide_playlist_recommendations_from_home_page", { status: "enable" | "disable" }>;
+        response: DataResponseMessage<"enable_hide_playlist_recommendations_from_home_page">;
+    };
```
Comment (VampireChicken12) [inline src/types/index.ts]: This is not necessary remove it.

**Review by VampireChicken12 — CHANGES_REQUESTED** [review]

File: src/types/index.ts (hunk 8 above)
Comment (HelloIamarandomperson) [reply src/types/index.ts]: Ok then

**Comment by VampireChicken12** [comment]
@allcontributors please add @HelloIamarandomperson for code

**Comment by VampireChicken12** [comment]
@HelloIamarandomperson thank you for adding this feature. I'm not sure when it will be released in a new version.

**Comment by VampireChicken12** [comment]
Below is the extension built with this feature included for being loaded as a temporary/unpacked extension
## Release Artifacts
| File Name | SHA-256 Hash |
| :--- | :---: |
| [youtube-enhancer-v1.27.1-Chrome.zip](https://github.com/user-attachments/files/18241795/youtube-enhancer-v1.27.1-Chrome.zip) | 80568b26439bbd1c98fbb022636e57526cf3b6bc020681bf195fb39faa55d83e |
| [youtube-enhancer-v1.27.1-Edge.zip](https://github.com/user-attachments/files/18241796/youtube-enhancer-v1.27.1-Edge.zip) | 80568b26439bbd1c98fbb022636e57526cf3b6bc020681bf195fb39faa55d83e |
| [youtube-enhancer-v1.27.1-Firefox.zip](https://github.com/user-attachments/files/18241798/youtube-enhancer-v1.27.1-Firefox.zip) | 7a3e37554c906dab189f9b49f19371be7adf76508b9971afbe03031008a557fa |
@HelloIamarandomperson @G-Ran-Berg

**Pull request merged**